crates/revmc-worker/src/external.rs

```diff
@@ -1,51 +1,70 @@
-use std::sync::{Arc, RwLock};
+use std::{
+    fmt::{Debug, Display},
+    num::NonZeroUsize,
+    sync::{Arc, RwLock},
+};
 
 use crate::{
     error::ExtError,
     worker::{aot_store_path, CompileWorker, SledDB},
 };
 use alloy_primitives::B256;
+use lru::LruCache;
 use once_cell::sync::OnceCell;
-use revm_primitives::SpecId;
+use revm::Database;
+use revm_primitives::{AccessListItem, SpecId};
 use revmc::EvmCompilerFn;
 
 pub(crate) static SLED_DB: OnceCell<Arc<RwLock<SledDB<B256>>>> = OnceCell::new();
 
 #[derive(Debug)]
-pub struct EXTCompileWorker {
+pub struct EXTCompileWorker<DB> {
     compile_worker: Box<CompileWorker>,
+    pub cache: LruCache<String, (EvmCompilerFn, libloading::Library)>,
+    _marker: std::marker::PhantomData<DB>,
```

Comment:
it seems marker is not used.

Reply:
its a placeholder for using generics

Reply:
ah! i see. how about seperate statedb from external context.
use `preload_cache` instead of cache_load_access_list`, its argument use vector of bytecode hash.
```
        &mut self,
        access_list: Vec<AccessListItem>,
        mut state_db: DB,
```
-->
```
        &mut self,
        code_hashs: Vec<B256>
```
i think it is more simple and clean design to be independent with statedb(evm database)

Reply:
Looks good! let me change

Reply:
done